kubernetes/controller/internal/controller/component/component_controller.go

```diff
@@ -367,22 +345,29 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, component *v1alpha1.Co
 	return nil
 }
 
-func (r *Reconciler) DetermineEffectiveVersionFromLiveRepo(ctx context.Context, component *v1alpha1.Component,
-	session ocmctx.Session, liveRepo ocmctx.Repository,
-) (_ string, err error) {
-	liveComponentAccess, err := liveRepo.LookupComponent(component.Spec.Component)
-	if err != nil {
-		return "", fmt.Errorf("failed to lookup component: %w", err)
+// convertRepositorySpec converts a JSON repository spec to a typed spec.
+func (r *Reconciler) convertRepositorySpec(spec *apiextensionsv1.JSON) (runtime.Typed, error) {
```

Comment:
This is a very complicated way to write `r.OCMScheme.Decode(spec.Raw)` and imho it needs not a method, but just a usage of the function inline because we do not need a `nil` check for spec. Is there reason to believe we need to deal with nil specs?